src/write_queue.rs

```diff
@@ -22,7 +22,8 @@ use std::io;
 use std::collections::VecDeque;
 use std::rc::{Rc, Weak};
 use std::cell::RefCell;
-use futures::{self, task, Async, Future, Poll};
+use futures::{self, task, Async, Poll};
+use futures::future::Future;
```

Comment:
What is the purpose of importing `futures::future::Future` rather than `futures::Future`? It does not look like `futures::Future` is deprecated: https://github.com/alexcrichton/futures-rs/blob/0.1.12/src/lib.rs#L177.